core/core-service-discovery/src/main/java/ai/wanaku/core/service/discovery/WanakuServiceRegistry.java

```diff
@@ -0,0 +1,118 @@
+package ai.wanaku.core.service.discovery;
+
+import ai.wanaku.api.types.management.Configuration;
+import ai.wanaku.api.types.management.Configurations;
+import ai.wanaku.api.types.management.Service;
+import ai.wanaku.api.types.management.State;
+import ai.wanaku.core.mcp.providers.ServiceRegistry;
+import ai.wanaku.core.mcp.providers.ServiceTarget;
+import ai.wanaku.core.mcp.providers.ServiceType;
+import ai.wanaku.core.persistence.api.ServiceRepository;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@ApplicationScoped
+public class WanakuServiceRegistry implements ServiceRegistry {
+
+    @Inject
+    ServiceRepository serviceRepository;
+
+    @Override
+    public void register(ServiceTarget serviceTarget, Map<String, String> configurations) {
+        Service service = serviceRepository.findById(serviceTarget.getService());
+        if (service == null) {
+            service = new Service();
+
+            service.setName(serviceTarget.getService());
+            service.setTarget(serviceTarget.toAddress());
+            service.setServiceType(serviceTarget.getServiceType().asValue());
+            Map<String, Configuration> configurationMap = new HashMap<>();
+            for (Map.Entry<String, String> entry : configurations.entrySet()) {
+                Configuration configuration = new Configuration();
+                configuration.setDescription(entry.getValue());
+                configurationMap.put(entry.getKey(), configuration);
+            }
+            Configurations config = new Configurations();
+            config.setConfigurations(configurationMap);
+            service.setConfigurations(config);
+
+            serviceRepository.persist(service);
```

Comment:
For now, I'd like to keep the services non-persistent. Previously (i.e.; up to 0.0.2) we used to persist then, but now we have auto-registration (using Valkey as the source of state) ... so let's keep them out of persistence for now.